src/twisted/web/tap.py

```diff
@@ -194,18 +213,40 @@ def postOptions(self):
         If no server port was supplied, select a default appropriate for the
         other options supplied.
         """
+        if self['port']:
+            def opt_port():
+                """
+                Dummy function to deprecate about
+                """
+            msg = deprecate.getDeprecationWarningString(opt_port,
+                                incremental.Version('Twisted', 17, 10, 0))
+            warnings.warn(msg, category=DeprecationWarning, stacklevel=2)
+            self['ports'].append(self['port'])
         if self['https']:
+            def opt_https():
+                """
+                Dummy function to deprecate about
+                """
+            msg = deprecate.getDeprecationWarningString(opt_https,
+                                incremental.Version('Twisted', 17, 10, 0))
+            warnings.warn(msg, category=DeprecationWarning, stacklevel=2)
             try:
                 reflect.namedModule('OpenSSL.SSL')
             except ImportError:
                 raise usage.UsageError("SSL support not installed")
-        if self['port'] is None:
+            sslStrport = 'ssl:port={}:privateKey={}:certKey={}'.format(
+                             self['https'],
+                             self['privkey'],
+                             self['certificate'],
+                         )
+            self['ports'].append(sslStrport)
+        if len(self['ports']) == 0:
             if self['personal']:
                 path = os.path.expanduser(
                     os.path.join('~', distrib.UserDirectory.userSocketName))
-                self['port'] = 'unix:' + path
```

Comment:
Is this a backwards incompatible change?

Reply:
It should be.